src/poetry/utils/env.py

```diff
@@ -341,7 +341,7 @@ def remove_distribution_files(self, distribution_name: str) -> list[Path]:
                     file.unlink()
 
             if distribution._path.exists():
-                shutil.rmtree(str(distribution._path))
+                remove_directory(str(distribution._path), force=True)
```

Comment:
Any reason to cast to string?

Reply:
It's a simple path, which mypy does not like.